### DISCUSSION TITLE
How to write a data attribute variant for data flags

### DISCUSSION THREAD

**Discussion by julian-amplica** [tailwindlabs/tailwindcss#12183] [Help] [answered]
I understand for a data attribute like `data-open="closed"` we can write `data-[open=closed]:...` to selectively apply tailwind classes, but what about a data attribute that is a flag—such as `data-disabled`—where there is no assigned value?

`radix-ui` uses lots of these flags (e.g., `data-placeholder` in the [`Select` component](https://www.radix-ui.com/primitives/docs/components/select)):

```html
<button ... data-state="closed" data-disabled> 
  ... 
</button>
```

I read the [data attribute variants documentation](https://tailwindcss.com/blog/tailwindcss-v3-2#data-attribute-variants), but it doesn't explain how to do this. 

Any help is much appreciated, thanks a lot!

**Comment by wongjn** [comment, marked as answer]
You could do `data-[disabled]:utility`, `data-[placeholder]:utility`, etc. See https://play.tailwindcss.com/ATlNeIi1dr

**Reply by julian-amplica** [reply]
Duh 🤦 Thanks for the help!
